### PR TITLE
feat: Stream support jamba

### PR DESCRIPTION
**Pull request by Josephasafg**

### CODE REVIEW

**Review by miri-bar — COMMENTED** [review]
Great!!!
Please see my small comments :)

File: ai21/clients/studio/resources/studio_resource.py [hunk 1]
```diff
@@ -14,23 +19,66 @@ def _post(
...
+    ) -> ResponseT | StreamT | None:
+        if stream and stream_cls is not None:
+            cast_to = extract_type(stream_cls)
+
+            return stream_cls(
+                cast_to=cast_to,
+                response=response,
+            )
+
+        if response_cls is not None:
```
[the remark this answers is not in the document]
Comment (Josephasafg) [reply ai21/clients/studio/resources/studio_resource.py:71]: Done

File: ai21/http_client.py [hunk 2]
```diff
@@ -130,14 +133,15 @@ def _request(
...
-        return self._client.request(
+        request = self._client.build_request(
             method=method,
             url=url,
             headers=headers,
             data=data,
             timeout=timeout,
             files=files,
         )
+        return self._client.send(request=request, stream=stream)
```
[the remark this answers is not in the document]
Comment (Josephasafg) [reply ai21/http_client.py:144]: Done

File: ai21/http_client.py [hunk 3]
```diff
@@ -99,23 +100,25 @@ def execute_http_request(
...
-            return self._client.request(
-                method=method,
-                url=url,
-                headers=headers,
-                timeout=timeout,
-                params=params,
+            request = self._client.build_request(
+                method=method, url=url, headers=headers, timeout=timeout, params=params
             )
+            return self._client.send(request=request, stream=stream)
```
[the remark this answers is not in the document]
Comment (Josephasafg) [reply ai21/http_client.py:121]: Done

File: ai21/types.py [hunk 4]
```diff
@@ -1,4 +1,14 @@
...
+from typing import Any, Union, List
+
+import httpx
+from typing_extensions import Literal, TypeVar, TYPE_CHECKING
+from ai21.stream import Stream
+
+if TYPE_CHECKING:
+    from ai21.models.ai21_base_model_mixin import AI21BaseModelMixin  # noqa
+
+ResponseT = TypeVar("_ResponseT", bound=Union["AI21BaseModelMixin", str, httpx.Response, List[Any]])
```
[the remark this answers is not in the document]
Comment (Josephasafg) [reply ai21/types.py:10]: Fixed. Left `str` and supported it

**Review by miri-bar — APPROVED** [review]

**Pull request merged**